sql-plugin/src/main/330+/scala/com/nvidia/spark/rapids/shims/GpuIntervalUtils.scala

```diff
@@ -557,4 +560,203 @@ object GpuIntervalUtils extends Arm {
       }
     }
   }
+
+  /**
+   * Cast day-time interval to string
+   * Rewrite from org.apache.spark.sql.catalyst.util.IntervalUtils.toDayTimeIntervalString
+   *
+   * @param micros     long micro seconds
+   * @param startField start field, valid values are [0, 3] indicates [DAY, HOUR, MINUTE, SECOND]
+   * @param endField   end field, should >= startField,
+   *                   valid values are [0, 3] indicates [DAY, HOUR, MINUTE, SECOND]
+   * @return ANSI day-time interval string, e.g.: interval '01 08:30:30.001' DAY TO SECOND
+   */
+  def toDayTimeIntervalString(
+      micros: ColumnVector,
+      startField: Byte,
+      endField: Byte): ColumnVector = {
+
+    val numRows = micros.getRowCount
+    val from = DT.fieldToString(startField).toUpperCase
+    val to = DT.fieldToString(endField).toUpperCase
+    val prefixStr = "INTERVAL '"
+    val postfixStr = s"' ${if (startField == endField) from else s"$from TO $to"}"
+
+    val retCv = withResource(new ArrayBuffer[ColumnVector]) { restHolder =>
+      // `restHolder` only hold one rest Cv;
+      // use `resetRest` to close the old one and set a new one
+      // make a copy of micros
+      restHolder += micros.incRefCount()
+
+      withResource(new ArrayBuffer[ColumnView]) { parts =>
+        // prefix part: INTERVAL '
+        parts += getConstStringVector(prefixStr, numRows)
+
+        // sign part: - or empty
```

Comment:
We may merge the sign vector and the prefix vector into one vector, to reduce the vectors to be concatenated.
```
            withResource(getConstStringVector("INTERVAL '-", numRows)) { neg =>
              withResource(getConstStringVector("INTERVAL '", numRows)) { empty =>
```

Reply:
done